test/testcne/ring_test.c

```diff
@@ -108,6 +114,97 @@ ring_name_tests(void *args)
     return 0;
 }
 
+static int
+test_ring_init(void *args)
+{
+    unsigned int ring_size, i;
+    struct ring_test_info tst = {0};
+    ssize_t tsize;
+    uint32_t bufcnt, bufsz;
+    size_t sz;
+
+    CNE_SET_USED(args);
+
+    tst.tst   = tst_start("Ring init API");
+    ring_size = cne_align32pow2(rand() % 16384);
+    tst_ok("Using ring_size=%zu\n", ring_size);
+
+    tsize = cne_ring_get_memsize_elem(0, ring_size);
+    TST_ASSERT_AND_CLEANUP(tsize > 0, "Ring memsize < 0\n", test_ring_cleanup, &tst);
+
+    tst.mm = mmap_alloc((tsize / CNE_CACHE_LINE_SIZE), CNE_CACHE_LINE_SIZE, MMAP_HUGEPAGE_4KB);
+    TST_ASSERT_AND_CLEANUP(tst.mm != NULL, "Out of memory\n", test_ring_cleanup, &tst);
+
+    sz = mmap_size(tst.mm, &bufcnt, &bufsz);
+    CNE_INFO("tsize: %ld, mmap size %ld, bufcnt %d, bufsz %d\n", tsize, sz, bufcnt, bufsz);
+
+    TST_ASSERT_AND_CLEANUP(sz >= ring_size, "mmap size too small\n", test_ring_cleanup, &tst);
+
+    tst.r = cne_ring_init(mmap_addr(tst.mm), 0, "ring init", 8, ring_size, 0);
+    TST_ASSERT_AND_CLEANUP(tst.r == NULL, "Ring invalid size should have failed\n",
+                           test_ring_cleanup, &tst);
+
+    tst.r = cne_ring_init(mmap_addr(tst.mm), sz - (4 * 1024), "ring init", 8, ring_size, 0);
+    TST_ASSERT_AND_CLEANUP(tst.r == NULL, "Ring invalid size should have failed\n",
+                           test_ring_cleanup, &tst);
+
+    /* Make address non-cacheline aligned */
+    tst.r = cne_ring_init(CNE_PTR_ADD(mmap_addr(tst.mm), 1), 0, "ring init", 8, ring_size, 0);
+    TST_ASSERT_AND_CLEANUP(tst.r == NULL, "Ring invalid address should have failed\n",
+                           test_ring_cleanup, &tst);
+
+    tst.r = cne_ring_init(mmap_addr(tst.mm), sz, "ring init", 8, ring_size, 0);
+    TST_ASSERT_AND_CLEANUP(tst.r != NULL, "Ring create failed\n", test_ring_cleanup, &tst);
+
+    if (verbose) {
+        vt_color(VT_BLUE, VT_NO_CHANGE, VT_OFF);
+        cne_ring_dump(NULL, tst.r);
+        cne_printf("\n");
+        vt_color(VT_DEFAULT_FG, VT_NO_CHANGE, VT_OFF);
+    }
+
+    TST_ASSERT_AND_CLEANUP(1 == cne_ring_empty(tst.r), "Ring empty expected\n", test_ring_cleanup,
+                           &tst);
+    TST_ASSERT_AND_CLEANUP(0 == cne_ring_full(tst.r), "Ring full unexpected\n", test_ring_cleanup,
+                           &tst);
+    TST_ASSERT_AND_CLEANUP(ring_size == cne_ring_get_size(tst.r),
+                           "ring size=%d expected instead of %d\n", test_ring_cleanup, &tst,
+                           ring_size - 1, cne_ring_get_size(tst.r));
+
+    for (i = 0; i < ring_size / 2; i++)
+        TST_ASSERT_AND_CLEANUP(0 == cne_ring_enqueue(tst.r, (void *)((uintptr_t)i + 1)),
+                               "Enqueue i=%u", test_ring_cleanup, &tst, i + 1);
+
+    TST_ASSERT_AND_CLEANUP(i == cne_ring_count(tst.r),
+                           "Ring count=%d mismatch with enqueued=%d elements\n", test_ring_cleanup,
```

Comment:
since ring_size and i are both unsigned ints, should we replace %d with %u when ring_size and i are being printed?

Reply:
I did not change this code in my patch, but I will update it.